statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DefinitionProcessor.java

```diff
@@ -79,20 +79,21 @@ private static int getNumDefinitionsInModel(final ResourceModel model) {
      * @param metricsRecorder The MetricsRecorder object used for collecting anonymous property usage metrics
      */
     public static void processDefinition(final AmazonWebServicesClientProxy proxy, final ResourceModel model, final MetricsRecorder metricsRecorder) {
-        if (model.getDefinitionS3Location() != null) {
-            model.setDefinitionString(fetchS3Definition(model.getDefinitionS3Location(), proxy, metricsRecorder));
+        if (model.getDefinitionString() != null && model.getDefinitionSubstitutions() != null) {
```

Comment:
Thought (general, blocking): are we able to reduce some of the duplicated logic? Is there a common pattern that each of these definition formats can follow to have their definition updated with the substitutions without adding that logic in three separate places?